MQ2Status.cpp

```diff
@@ -64,12 +64,15 @@ std::string to_string_with_precision(const T a_value, const int n = 6)
 // There currently isn't a way to check what lua stuff is running in c++ land
 // unlike plugins that use the g_pluginMap global
 // MQ2Lua uses a s_infoMap which is local to MQ2Lua
-static char* LuaScriptStatus(const char* scriptname) {
-	static char luaScript[64] = "";
+std::string LuaScriptStatus(const char* scriptname) {
+	std::ostringstream outputBuffer;
+	char luaScript[64] = "";
 	if (IsPluginLoaded("Lua")) {
 		// format is "${Lua.Script[scriptname].Status}"
 		sprintf_s(luaScript, "${Lua.Script[%s].Status}", scriptname);
 		ParseMacroData(luaScript, 64);
+		std::ostringstream out;
+		outputBuffer << luaScript;
```

Comment:
you do not need to use an ostringstream at all. You can convert to string directly without it.